### PR TITLE
Release packaging documentation

### LINKED ISSUE #960: Release documentation update
### What is your issue?

Update the release documentation

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)
 Added documentation details regarding packaging and releasing steps performed for OpenGHG release.
Also changed the position of packaging.rst so that it can be seen on website.

* **Please check if the PR fulfills these requirements**

- [x] Closes #960  (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Documentation and tutorials updated/added
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [x] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

File: doc/source/development/index_devel.rst [hunk 1]
```diff
@@ -25,6 +25,15 @@ Covers how we develop OpenGHG and the standards we adhere to
 
    python_devel
 
+Packaging & Release
+-------------------
+
+Covers steps to create a OpenGHG release and packaging steps
```
Comment (gareth-j) [inline doc/source/development/index_devel.rst:31]:
```suggestion
Covers OpenGHG packaging and release
```

File: doc/source/development/packaging.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+The development process should be:
+
+* New changes to any section of the code base should always be made using feature branches.
+
+.. code-block:: bash
+
+   git checkout -b branch_name
+
+.. note::
+   branch_name - Iss(Issue number) recommended if the change is a bug fix for an issue. E.g. “Iss123_add_attributes”. May also want to include “fix_” and “feature_” as appropriate.
```
Comment (gareth-j) [inline doc/source/development/packaging.rst:17]: I don't think we need this as a note. Maybe an example below with a `git checkout -b Iss961/updateReleaseDocs` or something.

File: doc/source/development/packaging.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
+==================
+Packaging releases
+==================
+
+OpenGHG is now completely tested and deployed using GitHub actions.
+The most updated developer code lies in the `OpenGHG repository <https://github.com/openghg/openghg>`_
+
+The development process should be:
```
Comment (gareth-j) [inline doc/source/development/packaging.rst:8]: Maybe this could be a numbered list of things?

File: doc/source/development/packaging.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+Releases aim to be backwards compatible and capable of being used for production runs, at least for
+the functionality that is fully described in the tutorial.
+
+We use `semantic versioning <https://semver.org>`__ and take care
+not to cause breaking changes in the public API.
+
+Creating a release
+------------------
+
+* The creation of the packages required by PyPI and conda is handled in the     GitHub actions workflow files under ``.github/workflows/workflow.yaml``. The package build and release workflow will only be triggered by a tagged commit. Before we get to this step we need to make sure everything is set up to do a release.
```
Comment (gareth-j) [inline doc/source/development/packaging.rst:39]: I don't think these need to be bullet points.

File: doc/source/development/packaging.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+* Ensure all issues in a ``milestone`` are complete
```
Comment (gareth-j) [inline doc/source/development/packaging.rst:41]:
```suggestion
Ensure all issues in the release milestone are complete
```

**Review by gareth-j — CHANGES_REQUESTED** [review]

**Comment by gareth-j** [comment]
So this needs a bit of work, I don't think we need to have the process as bullet points. The code/command examples look better when they're not breaking up a list of points.

**Comment by rt17603** [comment]
Was it possible to apply Gareth's suggested changes here (and then ask for re-review from myself)?

File: doc/source/development/packaging.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+version should not be used for production runs.
+
+Defining a release
+------------------
+
+Releases aim to be backwards compatible and capable of being used for production runs, at least for
+the functionality that is fully described in the tutorial.
+
+We use `semantic versioning <https://semver.org>`__ and take care
+not to cause breaking changes in the public API.
```
Comment (brendan-m-murphy) [inline doc/source/development/packaging.rst:34]: The semvar website says we should not have version numbers starting with 0. I guess it's not the most important part of what they specify

File: doc/source/development/packaging.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+Creating a release
+------------------
+
+* The creation of the packages required by PyPI and conda is handled in the     GitHub actions workflow files under ``.github/workflows/workflow.yaml``. The package build and release workflow will only be triggered by a tagged commit. Before we get to this step we need to make sure everything is set up to do a release.
+
+Ensure all issues in the release milestone are complete
```
Comment (brendan-m-murphy) [inline doc/source/development/packaging.rst:41]:
```suggestion
Ensure all issues in the release milestone are complete.
```

File: doc/source/development/packaging.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+* Now update the [Unreleased] section of the ``Changelog.md`` with the "release number - date of release". Ensure all the changes are included in the file. If not update it accordingly.
+
+* Also add [Unreleased] section at the top of ``Changelog.md`` along with the link to the latest diff as shown below:
+
+.. code-block:: bash
+
+   ## [Unreleased](https://github.com/openghg/openghg/compare/0.8.0...HEAD)
+
+   ## [0.8.0] - 2024-03-19
+
```
Comment (brendan-m-murphy) [inline doc/source/development/packaging.rst:61]: Are these changes committed to devel?

File: doc/source/development/packaging.rst [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+that is included in the package to tag versions.
+
+This will also trigger a full CI/CD to test and build the new version.
+Again, it should work as this tag was taken from your fully-tested
+"devel" branch.
+
+Completing the release
+----------------------
+
+Once you are satisfied that all workflows have run successfully check the latest version of the ``OpenGHG`` is live on both `PyPi <https://pypi.org/project/openghg/>`_ as well as `Anaconda <https://anaconda.org/openghg/openghg>`_.
```
Comment (brendan-m-murphy) [inline doc/source/development/packaging.rst:94]:
```suggestion
Once you are satisfied that all workflows have run successfully, check that the latest version of the ``OpenGHG`` is live on both `PyPi <https://pypi.org/project/openghg/>`_ as well as `Anaconda <https://anaconda.org/openghg/openghg>`_.
```

File: doc/source/packaging.rst [hunk 10]
```diff
@@ -3,19 +3,22 @@ Packaging releases
 ==================
```
Comment (brendan-m-murphy) [inline doc/source/packaging.rst:3]: This has a lot of overlap with development/packaging.rst. Do we need both?

Comment (rt17603) [reply doc/source/packaging.rst:3]: I noted that as well. Also, when I try and build the documentation I see:
```
/home/rt17603/openghg/doc/source/packaging.rst: WARNING: document isn't included in any toctree
```
so this doesn't seem to be included in the documentation build at the moment anyway.

I think any useful details from this should be incorporated into `development/packaging.rst` and this file should be deleted.

**Review by rt17603 — COMMENTED** [review]
I second the comment that we shouldn't need both (1) `doc/source/packaging.rst` and (2) `doc/source/development/packaging.rst` and it looks like (1) isn't currently included in the toctree anyway. My suggestion would be to incorporate any additional details from (1) into (2) and delete (1).

File: doc/source/development/packaging.rst (hunk 6 above)
Comment (SutarPrasad) [reply doc/source/development/packaging.rst:34]: Yes, we can decide on it sometime

File: doc/source/development/packaging.rst (hunk 8 above)
Comment (SutarPrasad) [reply doc/source/development/packaging.rst:61]: its on master

File: doc/source/packaging.rst (hunk 10 above)
Comment (SutarPrasad) [reply doc/source/packaging.rst:3]: done

File: doc/source/development/packaging.rst [hunk 11]
```diff
@@ -0,0 +1,89 @@
+==================
+Packaging releases
+==================
+
+OpenGHG is now fully tested and deployed using GitHub actions.
+The most updated developer code lies in the `OpenGHG repository <https://github.com/openghg/openghg>`_
+
+The development process should be;
```
Comment (brendan-m-murphy) [inline doc/source/development/packaging.rst:8]:
```suggestion
The development process should be:
```

Comment (brendan-m-murphy) [reply doc/source/development/packaging.rst:8]: I think you want to use ":" to start a list, rather than ";"

**Review by brendan-m-murphy — APPROVED** [review]
Looks good! I suggested one change that should be able to commit inline, but it's not a big deal.

**Review by brendan-m-murphy — APPROVED** [review]

**Pull request merged**